Type error: PageComponent/LayoutComponent is declared but never used

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.0.0
      eslint-config-next: 13.0.0
      react: 18.2.0
      react-dom: 18.2.0

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

When adding `"noUnusedLocals": true` to tsconfig.json and running a type-check with `npx tsc` (or `next build`), it fails due to declared but unused types:

```
.next/types/app/layout.ts:17:6 - error TS6196: 'PageComponent' is declared but never used.

17 type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>
        ~~~~~~~~~~~~~

.next/types/app/page.ts:18:6 - error TS6196: 'LayoutComponent' is declared but never used.

18 type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>
        ~~~~~~~~~~~~~~~
```

### Expected Behavior

No type errors.

### Link to reproduction

https://github.com/richardkall/next-bug-repro

### To Reproduce

1. Clone repo.
2. Run `npm install`
3. Run `npm run build`

--- a/packages/next/build/webpack/plugins/flight-types-plugin.ts
+++ b/packages/next/build/webpack/plugins/flight-types-plugin.ts
@@ -35,8 +35,8 @@ interface LayoutProps {
   params: any
 }
 
-type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>
-type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>
+export type PageComponent = (props: PageProps) => React.ReactNode | Promise<React.ReactNode>
+export type LayoutComponent = (props: LayoutProps) => React.ReactNode | Promise<React.ReactNode>
 
 interface IEntry {
   ${

